torchdata/dataloader2/reading_service.py

```diff
@@ -247,6 +250,19 @@ def initialize(self, datapipe: DataPipe) -> DataPipe:
                 for res_queue in res_queues:
                     res_queue.cancel_join_thread()
 
+        # Find replicable branches for worker processes
+        # The rest of non-replicable DataPipes will remain in the main process
+        replicable_dps = _find_replicable_branches(graph)
+        assert (
```

Comment:
Also a n00b question for my understanding, we can have multiple non-replicable DP, with single replicable branch in between, e.g. non-replicable -> map_dp -> fullsync?

Reply:
Yeah